pkg/kubelet/metrics/metrics.go

```diff
@@ -736,6 +739,36 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+
+	// This metrics is still valid across Kubelet restart
+	KubeletBootstrapDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           KubeletBootstrapKey,
+			Help:           "Duration in seconds from kueblet running to first-time attempting node registeration",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// This metrics shouldn't be reported across Kubelet restart
+	NodeRegistrationDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           NodeRegistrationKey,
+			Help:           "Duration in seconds from first-time attempting node registeration to registered new node",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
+	// This metrics shouldn't be reported across Kubelet restart
+	NodeReadyDuration = metrics.NewGauge(
+		&metrics.GaugeOpts{
+			Subsystem:      KubeletSubsystem,
+			Name:           NodeReadyKey,
+			Help:           "Duration in seconds from registered new node to first-time node ready",
```

Comment:
Does it make sense to measure all NodeReady duration from the start of kubelet?

The reason is that kubelet does many things in parallel. The works that make the node Ready don't necessarily start after the node is registered. So it might make more sense to cacluate the duration from the start of kubelet.

Reply:
I think they are mathematically equivalent. Usability/Understandability may matter most here. Say there are timestamp A -> B -> C -> D.

Option 1: A -> B, B -> C, C -> D
Option 2: A -> D, A -> C, A -> D

I don't have strong preference. Let's discuss more.